Dolphin/windWakerInterface.py

```diff
@@ -263,6 +267,18 @@ def take_pearl(item: int):
     if curr_val == 0x7:
         toggle_bit_flag(0x803C524A, 6, False)
 
+def give_magic_upgade():
+    dme.write_byte(0x803C4C1B, 0x20)
+    dme.write_byte(0x803C4C1C, 0x20)
+
+def take_magic_upgade():
+    dme.write_byte(0x803C4C1B, 0x10)
+
+def give_hurricane_spin():
+    dme.write_byte(0x803C5295, 1)
```

Comment:
This is looking like it's actually a single-bit toggle. Can you please rewrite it to be more like the `give_heros_charm` and `take_heros_charm` below? It should just be an offset of `0`

Reply:
Also requesting a change for the magic or only for hurricane spin?

Reply:
Only the Hurricane Spin. Magic Meter Upgrade is fine.